include/boost/math/statistics/univariate_statistics.hpp

```diff
@@ -512,6 +512,74 @@ inline auto interquartile_range(RandomAccessContainer & v)
     return interquartile_range(v.begin(), v.end());
 }
 
+template<class ForwardIterator, class OutputIterator>
+auto sorted_mode(ForwardIterator first, ForwardIterator last, OutputIterator output) -> decltype(output)
+{
+    using Z = typename std::iterator_traits<ForwardIterator>::value_type;
+    static_assert(std::is_integral<Z>::value, "Floating point values have not yet been implemented.");
+    using Size = typename std::iterator_traits<ForwardIterator>::difference_type;
+
+    std::vector<Z> modes {};
+    modes.reserve(16);
+    Size max_counter {0};
+    auto it {first};
+
+    if(first == last)
+    {
+        return output;
+    }
+    
+    while(it != last)
+    {
+        Size current_count {0};
+        auto end_it {it};
+        while(end_it != last && *end_it == *it)
+        {
+            ++current_count;
+            ++end_it;
+        }
+
+        if(current_count > max_counter)
+        {
+            modes.resize(1);
+            modes[0] = *it;
+            max_counter = current_count;
+        }
+
+        else if(current_count == max_counter)
+        {
+            modes.emplace_back(*it);
+        }
+
+        it = end_it;
+    }
+
+    if(max_counter == 1) // Return an empty vector if mode is undefined
```

Comment:
This is not undefined, it's just GIGO. If the input is `{ 99 }` then the mode is `99`, that's it. 

And once you remove this edge case for size 1, then you can _also_ remove the special case for `first == last` because the `std::move` will be a no-op in that case.